python_modules/dagster-graphql/dagster_graphql/implementation/fetch_assets.py

```diff
@@ -377,6 +378,60 @@ def get_materialized_partitions_subset(
         )
 
 
+def get_failed_partitions_subset(
+    instance: DagsterInstance,
+    asset_key: AssetKey,
+    dynamic_partitions_loader: DynamicPartitionsStore,
+    partitions_def: Optional[PartitionsDefinition] = None,
+) -> Optional[PartitionsSubset]:
+    """
+    Returns the materialization status for each partition key. The materialization status
+    is a boolean indicating whether the partition has been materialized: True if materialized,
+    False if not.
+    """
+    if not partitions_def:
+        return None
+
+    if instance.can_cache_asset_status_data() and isinstance(
+        partitions_def, CACHEABLE_PARTITION_TYPES
+    ):
+        # When the "cached_status_data" column exists in storage, update the column to contain
+        # the latest partition status values
+        updated_cache_value = get_and_update_asset_status_cache_value(
+            instance, asset_key, partitions_def
+        )
+        materialized_subset = (
+            updated_cache_value.deserialize_failed_partition_subsets(partitions_def)
+            if updated_cache_value
+            else partitions_def.empty_subset()
+        )
+
+        return materialized_subset
+
+    else:
+        # If the partition status can't be cached, fetch partition status from storage
```

Comment:
maybe we should just return an empty subset instead?

Reply:
depends on how fast the query runs--I'd be curious if we can test this to figure out if the query is slow without the caching. but I'm fine with not supporting the failed partitions functionality if the user hasn't run the migration

Reply:
I'm also fine with that

Reply:
I'll stick with the approach we used for materializations for now